src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -77,18 +77,24 @@ def select_interpreter_for_targets(self, targets):
     """Pick an interpreter compatible with all the specified targets."""
     tgts_by_compatibilities = defaultdict(list)
     filters = set()
-    for target in targets:
-      if isinstance(target, PythonTarget):
-        c = self._python_setup.compatibility_or_constraints(target)
-        tgts_by_compatibilities[c].append(target)
-        filters.update(c)
+
+    if self._python_setup.get_options().is_flagged('interpreter_constraints'):
+      # Add the last option passed down in the option constraints.
```

Comment:
I think this comment wording was my suggestion and it can be improved. Maybe something like: 
`# Pants will append the CLI-supplied constraints after the pants.ini constraints and we want to take only the CLI-supplied constraints in this case.`